src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -451,26 +455,52 @@ dnnl::post_ops program_node::try_optimize_post_ops(dnnl::post_ops& p_ops, const
             case onednn_post_op_type::optimized_eltwise_clip: return onednn_post_op_type::eltwise_clip;
             case onednn_post_op_type::optimized_eltwise_linear: return onednn_post_op_type::eltwise_linear;
             case onednn_post_op_type::optimized_eltwise_round: return onednn_post_op_type::eltwise_round;
+            case onednn_post_op_type::optimized_sum: return onednn_post_op_type::sum;
             default:
                 throw std::runtime_error("Unsupported optimized eltwise post-operation type");
                 break;
         }
     };
 
+    auto remove_optimized_prefix = [&](std::vector<fused_primitive_desc_onednn>& post_ops) {
+        // Check and update post-op map if we already optimized something
+        auto iter = post_ops.begin();
+        for (; iter != post_ops.end(); ) {
+            if (type_is_optimized_sum(iter->op_type)) {
+                iter->op_type = onednn_post_op_type::sum;
+                ++iter;
+            } else if (type_is_optimized_eltwise(iter->op_type)) {
+                iter->op_type = get_eltwise_type(iter->op_type);
+                ++iter;
+            } else if (type_is_optimized(iter->op_type)) {
+                iter = post_ops.erase(iter);
+            } else {
+                ++iter;
+            }
+        }
+    };
+
     auto& cur_post_ops = get_fused_primitives_onednn();
 
     size_t cur_post_op_idx = 1;
     size_t prev_post_op_idx = 0;
     bool optimization_done = false;
 
-    // Check and update post-op map if we already optimized something
-    for (size_t post_op_idx = 0; post_op_idx < cur_post_ops.size(); post_op_idx++) {
-        if (type_is_optimized_sum(cur_post_ops[post_op_idx].op_type))
-            cur_post_ops[post_op_idx].op_type = onednn_post_op_type::sum;
-        else if (type_is_optimized_eltwise(cur_post_ops[post_op_idx].op_type))
-            cur_post_ops[post_op_idx].op_type = get_eltwise_type(cur_post_ops[post_op_idx].op_type);
-        else if (type_is_optimized(cur_post_ops[post_op_idx].op_type))
-            cur_post_ops.erase(cur_post_ops.begin() + post_op_idx);
+    GPU_DEBUG_IF(debug_config->verbose >= 3) {
```

Comment:
Thanks for adding related debug info! :)

Reply:
It was Mingyu's idea.